src/libstd/panicking.rs

```diff
@@ -12,11 +12,12 @@ use core::panic::{BoxMeUp, PanicInfo, Location};
 use crate::any::Any;
 use crate::fmt;
 use crate::intrinsics;
+use crate::lock_api::RawRwLock as _;
 use crate::mem;
+use crate::parking_lot::RawRwLock;
```

Comment:
So everything else uses std's `RawLock` but this uses the one from `parking_lot`... why that?

Reply:
Oh, that's because it needs the `RwLock`, not the `Mutex`, I guess.

Seems like another agument to make the rest also use parking_lot's stuff directly, though.

Reply:
But they are not equivalent. See my other comment. `parking_lot::RawMutex` is lower level and would require a lot of manual unlocking where we now get that much nicer with the guard. The standard library had a mutex with a guard prior to this PR. So going for the `parking_lot` stuff directly for the mutexes would make everything more low level and harder to get fully correct.

Reply:
The previous `crate::sys_common::rwlock::RWLock` did not have any guards. But the `crate::sys_common::mutex::Mutex` did. So doing what this PR does currently keeps libstd as similar as possible. Going directly for `parking_lot.:RawMutex` would be a step backwards IMO. We could however write a small wrapper for the RwLock that has guards. But given it's only used in so few places that felt a bit overkill.

Reply:
Okay, so this one file here is where we are okay with not having guards?

Reply:
I suppose so. It's how the standard library is currently implemented. So someone must have thought it was okay at some point. Well, here and in one place SGX also uses `parking_lot::RawRwLock` directly in this PR.  I would not be against introducing a simple `sync::RawRwLock` with guards. But it would not be that efficient to use in the SGX place, so it would still only be `panicking.rs` that would use it.